src/OpenLoco/Vehicles/VehicleBody.cpp

```diff
@@ -948,7 +948,7 @@ namespace OpenLoco::Vehicles
 
         auto bogieDifference = frontBogie->position - backBogie->position;
 
-        auto smokeLoc = bogieDifference * var_05 / 128 + frontBogie->position + Map::map_pos3{ xyFactor.x, xyFactor.y, vehicleObject->animation[num].height };
+        auto smokeLoc = bogieDifference * var_05 / 128 + frontBogie->position + Map::map_pos3{ static_cast<coord_t>(xyFactor.x), static_cast<coord_t>(xyFactor.y), vehicleObject->animation[num].height };
```

Comment:
We may be able to avoid these casts. Perhaps we can instead use:
```suggestion
        auto smokeLoc = bogieDifference * var_05 / 128 + frontBogie->position + Map::map_pos3(xyFactor.x, xyFactor.y, vehicleObject->animation[num].height);
```

Similarly below.